ktoml-core/src/commonMain/kotlin/com/akuleshov7/ktoml/decoders/TomlAbstractDecoder.kt

```diff
@@ -25,6 +34,23 @@ public abstract class TomlAbstractDecoder : AbstractDecoder() {
     override fun decodeDouble(): Double = decodePrimitiveType()
     override fun decodeString(): String = decodePrimitiveType()
 
+    protected fun DeserializationStrategy<*>.isDateTime(): Boolean {
+        return descriptor == instantSerializer.descriptor
+                || descriptor == localDateTimeSerializer.descriptor
+                || descriptor == localDateSerializer.descriptor
+    }
+
+    // Cases for date-time types
+    @Suppress("UNCHECKED_CAST")
+    override fun <T> decodeSerializableValue(deserializer: DeserializationStrategy<T>): T {
+        return when (deserializer.descriptor) {
+            instantSerializer.descriptor -> decodePrimitiveType<Instant>() as T
```

Comment:
Why do we need this workaround here? To cast everything to `T`?

Reply:
`decodePrimitiveType<Instant>()` returns an `Instant`, where `decodeSerializableValue` expects a T. We know that `Instant` is correct because of the serializer but we still need to cast (this code is inspired by the doc at https://github.com/Kotlin/kotlinx.serialization/blob/master/docs/formats.md#format-specific-types)

Reply:
Yeah, that's what I thought